executor/sort.go

```diff
@@ -352,6 +360,40 @@ func (e *SortExec) spillToDiskByRowPtr() (disk *chunk.ListInDisk, err error) {
 	return rowChunksInDisk, nil
 }
 
+// ActionSpill returns a memory.ActionOnExceed for spilling over to disk.
+func (e *SortExec) ActionSpill() memory.ActionOnExceed {
+	return &spillSortDiskAction{e: e}
+}
+
+// spillSortDiskAction implements memory.ActionOnExceed for chunk.List. If
+// the memory quota of a query is exceeded, spillSortDiskAction.Action is
+// triggered.
+type spillSortDiskAction struct {
+	once           sync.Once
+	e              *SortExec
+	fallbackAction memory.ActionOnExceed
+}
+
+// Action sends a signal to trigger spillToDisk method of SortExec
+// and if it is already triggered before, call its fallbackAction.
+func (a *spillSortDiskAction) Action(t *memory.Tracker) {
+	if a.e.alreadySpilledSafe() {
+		if a.fallbackAction != nil {
+			a.fallbackAction.Action(t)
+		}
+	}
+	a.once.Do(func() {
```

Comment:
We may need a method called `spillSortDiskAction.reset` to reset `a.once` and `a.e.exceeded`.